### PR TITLE
Benchmark - Updating llama-2 requirement files

### PR DESCRIPTION
**Pull request by ironmangithub**

### CODE REVIEW

File: onnxruntime/python/tools/transformers/models/llama/requirements-cpu.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 -r requirements.txt
-onnxruntime>=1.16.2
\ No newline at end of file
+onnxruntime>=1.16.2
```
Comment (snnn) [inline onnxruntime/python/tools/transformers/models/llama/requirements-cpu.txt:2]: Where does this file get used?

**Comment by tianleiwu** [comment]
@kunal-vaishnavi, could you double-check whether those packages are needed for benchmark llama?

File: onnxruntime/python/tools/transformers/models/llama/requirements-cpu.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
...
+psutil
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/models/llama/requirements-cpu.txt:3]: If it is needed for both cpu and gpu, we can add psutil to requirements.txt

**Comment by kunal-vaishnavi** [comment]
> @kunal-vaishnavi, could you double-check whether those packages are needed for benchmark llama?

Both are used [here](https://github.com/microsoft/onnxruntime/blob/ec0e4d3b6572c18a3462eb6efb3bb007ec3a2962/onnxruntime/python/tools/transformers/models/llama/benchmark.py#L364) during benchmarking (`psutil` for CPU usage and `py3nvml` for measuring GPU memory usage).

**Comment by kunal-vaishnavi** [comment]
Since this PR is updating `requirements.txt` files, can we also [move](https://github.com/microsoft/onnxruntime/blob/ec0e4d3b6572c18a3462eb6efb3bb007ec3a2962/onnxruntime/python/tools/transformers/models/whisper/requirements.txt#L14) `py3nvml` to `requirements-cuda.txt` for Whisper and [move](https://github.com/microsoft/onnxruntime/blob/ec0e4d3b6572c18a3462eb6efb3bb007ec3a2962/onnxruntime/python/tools/transformers/models/whisper/requirements.txt#L12) `onnx>=1.15.0` before `protobuf==3.20.2` and `numpy==1.23.3` in `requirements.txt` for Whisper (to ensure that the right versions of `protobuf` and `numpy` are installed)?

File: onnxruntime/python/tools/transformers/models/llama/requirements-cpu.txt (hunk 1 above)
Comment (kunal-vaishnavi) [reply onnxruntime/python/tools/transformers/models/llama/requirements-cpu.txt:2]: These `requirements-*.txt` files are used in Anubis, Olive, and inference example notebooks. They also help users know exactly which package versions to install for their scenarios.

File: onnxruntime/python/tools/transformers/models/llama/requirements-cpu.txt (hunk 2 above)
Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/llama/requirements-cpu.txt]: done.

**Review by kunal-vaishnavi — DISMISSED** [review]

**Comment by kunal-vaishnavi** [comment]
It appears the "Big Models" CI pipeline is not using the right `requirements-*.txt` file when benchmarking. I think you can keep the [changes](https://github.com/microsoft/onnxruntime/pull/19716#issuecomment-1971759629) and then change the below code to use `requirements-cuda.txt` instead of `requirements.txt` since the benchmark is happening on CUDA.

https://github.com/microsoft/onnxruntime/blob/ed550b5fe5aa41e182db84d2b2f2fb768121fd7a/tools/ci_build/github/azure-pipelines/bigmodels-ci-pipeline.yml#L396-L422

**Comment by mszhanyi** [comment]
> It appears the "Big Models" CI pipeline is not using the right `requirements-*.txt` file when benchmarking. I think you can keep the [changes](https://github.com/microsoft/onnxruntime/pull/19716#issuecomment-1971759629) and then change the below code to use `requirements-cuda.txt` instead of `requirements.txt` since the benchmark is happening on CUDA.
> 
> https://github.com/microsoft/onnxruntime/blob/ed550b5fe5aa41e182db84d2b2f2fb768121fd7a/tools/ci_build/github/azure-pipelines/bigmodels-ci-pipeline.yml#L396-L422

Big models is used to check PR's correctness. So it uses the ort-gpu generated by current commit. So it could NOT use the requirements_cuda.txt

**Comment by kunal-vaishnavi** [comment]
> Big models is used to check PR's correctness. So it uses the ort-gpu wheel built by current commit. It could NOT use the requirements_cuda.txt which installs the released ort wheel

The released ORT GPU package can be uninstalled and then the ORT GPU package built by the current commit (`/ort-artifact/*.whl`) can be installed afterwards (just like how you currently uninstall `torch` for CPU and then install `torch` for GPU).

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**